.gitignore

```diff
@@ -97,3 +97,5 @@ gradle-app.setting
 # # Work around https://youtrack.jetbrains.com/issue/IDEA-116898
 # gradle/wrapper/gradle-wrapper.properties
 
+
+application.properties
```

Comment:
Generalnie lepszą praktyką jest nietrzymanie parametrów w repozytorium. Nie chodzi tu już o bezpieczeństwo danych i haseł (choć to też warto mieć na uwadze), a raczej o szczegółowość.

Załóżmy, że aplikacja łączy się Redisem, a każda instancja aplikacji ma osobną instancję Redisa. Szybko zauważamy, że nie da się tego załatwić takim plikiem z repozytorium. W zamian dla każdej instancji możemy dostarczyć osobny plik z parametrami przez 'provision'  - z użyciem playbooków ansible, tasków różnych menadżerów gulp, grunt (czy cokolwiek jest popularne w danym dniu w JS) albo prostych skryptów shellowych.